discovery-infra/assisted_service_api.py

```diff
@@ -14,9 +18,52 @@ def __init__(self, inventory_url):
         self.inventory_url = inventory_url
         configs = Configuration()
         configs.host = self.inventory_url + "/api/assisted-install/v1"
+        self.set_config_auth(configs)
+
         self.api = ApiClient(configuration=configs)
         self.client = api.InstallerApi(api_client=self.api)
 
+    def set_config_auth(self, c):
+        offline_token = os.environ.get('OFFLINE_TOKEN', "")
+        if offline_token == "": return
+
+        def refresh_api_key(config):
+            # Get the properly padded key segment
+            auth = config.api_key.get('Authorization', None)
+            if auth != None:
+                segment = auth.split('.')[1]
+                padding = len(segment) % 4
+                segment = segment + padding * '='
+
+                expires_on = json.loads(base64.b64decode(segment))['exp']
+
+                # if this key doesn't expire or if it has more than 30 seconds left, don't refresh
+                remaining = expires_on - time.time()
+                if expires_on == 0 or remaining > 30:
+                    return
+
+            # fetch new key if expired or not set yet
+            token_url = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+            env_url = os.environ.get("SSO_URL", "")
+            if env_url != "": token_url = env_url
+
+            params = {
+                "client_id":     "cloud-services",
+                "grant_type":    "refresh_token",
+                "refresh_token": offline_token,
+            }
+
+            log.info("Refreshing API key")
+            response = requests.post(token_url, data = params)
+            if not response.ok:
+                log.error("Failed to fetch JWT token: %: %" % response.reason, response.text)
```

Comment:
can we continue to run with this error?

Reply:
Probably not. Same argument as above.

Reply:
so i think u need to throw exception here

Reply:
u have such functionality in responce. check and throw exception on error